### PR TITLE
Plugins: add error notice explanation for a server error

### PR DESCRIPTION
**Pull request by eoigal**
Related to https://github.com/Automattic/wp-calypso/issues/92251

## Proposed Changes

* Adds an error notice explanation when the plugin install experiences a server error when activating the plugin
* Adds a partial success action to allow us to handle case where install was successful but the activation failed. This will allow the modal to reflect this by showing the installed (but deactivated) plugin.

https://github.com/Automattic/wp-calypso/assets/5560595/ee2ce058-9f9f-4982-bc6c-1bf971fe9e0b

## Before

Screenshot 2024-07-02 at 14 26 51

## After

Screenshot 2024-07-02 at 14 28 19
Screenshot 2024-07-03 at 18 29 42

## Why are these changes being made?

* Better explanation to user what happened when they tried to install a plugin

## Testing Instructions

1. Go to global plugins view at `/plugins/widont-part-deux`
2. Install the plugin on a WoA site
3. Observe the install and eventual error message
4. Go to `wp-admin` and verify that the plugin did install but did not activate.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/plugins/notices.jsx [hunk 1]
```diff
@@ -821,6 +821,14 @@ class PluginNotices extends Component {
 							}
 						);
 
+					case 'server_error':
+						return translate(
+							'Error activating %(plugin)s on %(site)s because it triggered a fatal error. As a safety measure, the plugin is installed but remains deactivated.',
```
Comment (Addison-Stavlo) [inline client/my-sites/plugins/notices.jsx]: does `server_error` always mean it was installed but not activated?

Comment (eoigal) [reply client/my-sites/plugins/notices.jsx]: I added a `lastStep` to track where the `errorCallback` occurred, if the server error happened during the `doActivate` then this error state should be accurate.

**Review by lupus2k — APPROVED** [review]
LGTM :tada: 

My only concern is the `Failed to install` message on the sites list

Before:
image

After:
image

**Comment by eoigal** [comment]
> My only concern is the `Failed to install` message on the sites list

I added another issue for this as I think there is a bug in how we handle the plugin status - it doesn't render consistently. I think we can handle this in a follow up.

https://github.com/Automattic/wp-calypso/issues/92396

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16076758

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @eoigal for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16076758).

**Pull request merged**